.circleci/config.yml

```diff
@@ -25,28 +46,24 @@ jobs:
     resource_class: xlarge
     steps:
       - checkout
-      - run: |
-            pip3 install requests
-            if [[ $(./scripts/needs_ci_run) == "skip CI" ]]; then
-                echo "Changes in last commit do not need CI. Skipping step"
-                circleci-agent step halt
-            fi
-      - run: sudo python3 -m pip install pre-commit && pre-commit run --all-files && ./scripts/error_on_dirty.sh
-      - install_linux_dependencies
+      - linux_preflight
+      - check_ci_lint
+      - linux_deps
       - run: git submodule sync && git submodule update --init
       - restore_cache:
           keys:
             - ccache-debug-linux-x86_64-v3-{{ .Branch }}-{{ checksum "conanfile.txt" }}
             - ccache-debug-linux-x86_64-v3-{{ checksum "conanfile.txt" }}
-      - run: mkdir build
       - run: |
+          mkdir build
           # NOTE: -Werror disabled in CI, as we currently have >4k warnings.
-          cd build \
-            && cmake .. -DCMAKE_BUILD_TYPE=Debug -DENABLE_COVERAGE=On -DCPACK_GENERATOR=DEB \
-                -DENABLE_COMPILER_WARNINGS=On -DENABLE_WERROR=Off -DCMAKE_EXPORT_COMPILE_COMMANDS=On \
-                -DCMAKE_CXX_FLAGS="-fuse-ld=lld" \
-                -DLOGGING_LEVEL=INFO \
-                -DENABLE_PYTHON_BINDINGS=On
```

Comment:
there was no real benefit to `&&`ing these together so i just put them on separate lines